Cesium3DTilesSelection/src/WebMapTileServiceRasterOverlay.cpp

```diff
@@ -0,0 +1,333 @@
+                            : std::nullopt;
+
+  bool hasError = false;
+  std::string errorMessage;
+
+  std::string layer;
+  std::string style;
+  std::string tileMatrixSetID;
+  if (!_options.layer.has_value()) {
+    hasError = true;
+    errorMessage = "layer is required.";
+  } else {
+    layer = _options.layer.value();
+  }
+  if (!_options.style.has_value()) {
+    hasError = true;
+    errorMessage = "style is required.";
+  } else {
+    style = _options.style.value();
+  }
+  if (!_options.tileMatrixSetID.has_value()) {
+    hasError = true;
+    errorMessage = "tileMatrixSetID is required.";
+  } else {
+    tileMatrixSetID = _options.tileMatrixSetID.value();
+  }
+  std::string format = _options.format.value_or("image/jpeg");
+  uint32_t tileWidth = _options.tileWidth.value_or(256);
+  uint32_t tileHeight = _options.tileHeight.value_or(256);
+
+  uint32_t minimumLevel = std::numeric_limits<uint32_t>::max();
+  uint32_t maximumLevel = 0;
+
+  if (maximumLevel < minimumLevel && maximumLevel == 0) {
+    // Min and max levels unknown, so use defaults.
+    minimumLevel = 0;
+    maximumLevel = 25;
+  }
+
+  bool useKVP;
+
+  std::regex bracket("\\{");
+  std::regex bracketS("\\{s\\}");
+  std::smatch matchBracket;
+  bool hasMatch = std::regex_search(_url, matchBracket, bracket);
+  if (!hasMatch ||
+      (matchBracket.size() == 1 && std::regex_match(_url, bracketS))) {
+    useKVP = true;
+  } else {
+    useKVP = false;
+  }
+
+  std::optional<std::map<std::string, std::string>> dimensions =
+      _options.dimensions;
+
+  minimumLevel = glm::min(minimumLevel, maximumLevel);
+
+  minimumLevel = _options.minimumLevel.value_or(minimumLevel);
+  maximumLevel = _options.maximumLevel.value_or(maximumLevel);
+
+  CesiumGeospatial::GlobeRectangle tilingSchemeRectangle =
+      CesiumGeospatial::GeographicProjection::MAXIMUM_GLOBE_RECTANGLE;
+  CesiumGeospatial::Projection projection;
+
+  if (_options.projection) {
+    projection = _options.projection.value();
+  } else {
+    if (_options.ellipsoid) {
+      projection =
+          CesiumGeospatial::WebMercatorProjection(_options.ellipsoid.value());
+    } else {
+      projection = CesiumGeospatial::WebMercatorProjection();
+    }
+  }
+  CesiumGeometry::Rectangle coverageRectangle =
+      _options.coverageRectangle.value_or(
+          projectRectangleSimple(projection, tilingSchemeRectangle));
+  CesiumGeometry::QuadtreeTilingScheme tilingScheme(coverageRectangle, 1, 1);
+
+  std::vector<std::string> subdomains;
+  if (_options.subdomains) {
+    const std::variant<std::string, std::vector<std::string>>& subdomains_v =
+        _options.subdomains.value();
+    if (subdomains_v.index() == 0) {
+      const std::string& subdomains_s = std::get<std::string>(subdomains_v);
+      for (std::string::const_iterator it = subdomains_s.begin();
+           it != subdomains_s.end();
+           ++it) {
+        subdomains.emplace_back(std::to_string(*it));
```

Comment:
I don't think this will work right. `to_string` will treat its parameter as a number, not a character. `subdomains.emplace_back(1, *it);` should do the trick I believe (by using the corresponding std::string constructor).